### PR TITLE
fix to bucket queries not returning content

### PR DESCRIPTION
**Pull request by joshLong145**
Fixes issue observed on chain where WhereIs data kept within the bucket keeper would not return in the response body.

## Changes
- refactor x/bucket/keeper to use `setWhereIs` over `AppendWhereIs`
- Added missing property assignments
- made pagination optional in whereIs handlers.

## Checklist

- [ ] Unit tests
- [ ] Documentation

## References *(optional)*

Fixes 
Connects 

<sub> Mention [stepsize] in a comment if you'd like to report some technical debt. See examples [here](https://app.stepsize.com/api/demo-pr-redirect).</sub>

### CODE REVIEW

File: x/bucket/keeper/where_is.go [hunk 1]
```diff
@@ -128,12 +114,13 @@ func (k Keeper) GetAllWhereIs(ctx sdk.Context) (list []types.WhereIs) {
 	for ; iterator.Valid(); iterator.Next() {
 		var val types.WhereIs
 		k.cdc.MustUnmarshal(iterator.Value(), &val)
+		fmt.Printf("current key for where_is in store: %s", val.Did)
```
Comment (ntindle) [inline x/bucket/keeper/where_is.go]: Should we drop the printlns

File: x/bucket/keeper/where_is.go [hunk 2]
```diff
@@ -128,12 +114,13 @@ func (k Keeper) GetAllWhereIs(ctx sdk.Context) (list []types.WhereIs) {
...
 		list = append(list, val)
 	}
 
 	return
 }
 
 func (k Keeper) GenerateKeyForDID() string {
-	return uuid.New().String()
+	return strings.Replace(uuid.New().String(), "-", "", -1)
```
Comment (ntindle) [inline x/bucket/keeper/where_is.go:125]: Nice save

File: x/bucket/types/messages_where_is.go [hunk 3]
```diff
@@ -19,7 +19,11 @@ var _ sdk.Msg = &MsgCreateWhereIs{}
 
 func NewMsgCreateWhereIs(creator string, label string, role BucketRole, visibility BucketVisibility, content []*BucketItem) *MsgCreateWhereIs {
 	return &MsgCreateWhereIs{
-		Creator: creator,
+		Creator:    creator,
```
Comment (ntindle) [inline x/bucket/types/messages_where_is.go:22]: Are these all the fields? I feel as if there were more

File: x/bucket/keeper/where_is.go [hunk 4]
```diff
@@ -94,24 +98,6 @@ func (k Keeper) GetWhereIsByCreator(ctx sdk.Context, creator string) (list []typ
 	return
 }
 
-// GetWhereIs returns a whereIs from its id
-func (k Keeper) GetWhereIsAll(ctx sdk.Context, creator, id string) (val types.WhereIs, found bool) {
```
Comment (ntindle) [inline x/bucket/keeper/where_is.go:98]: Why removed?

**Review by ntindle — CHANGES_REQUESTED** [review]

File: x/bucket/keeper/where_is.go (hunk 4 above)
Comment (joshLong145) [reply x/bucket/keeper/where_is.go:98]: it wasnt being used, duplicated method.

File: x/bucket/types/messages_where_is.go (hunk 3 above)
Comment (joshLong145) [reply x/bucket/types/messages_where_is.go:22]: we have acl properties but they are not being respected at this time so i dont allow them to be set

**Review by ntindle — APPROVED** [review]

**Review by 6missedcalls — APPROVED** [review]
LGTM

**Review by prdlk — APPROVED** [review]

**Review by prdlk — APPROVED** [review]

**Pull request merged**